pages/Livesite/announcements.js

```diff
@@ -1,24 +1,178 @@
-import React from 'react';
+/* eslint-disable no-alert */
+import React, { useState, useEffect } from 'react';
+import { format } from 'timeago.js';
+import ReactMarkdown from 'react-markdown';
+import AnnouncementsCard from '../../components/announcementsCard';
+import Modal from '../../components/modal';
+import TextBox from '../../components/textbox';
 import Page from '../../components/page';
-import { getHackathons } from '../../utility/firebase';
-import { LIVESITE_NAVBAR } from '../../constants';
-
-export default ({ hackathons }) => (
-  <Page
-    currentPath="Livesite"
-    hackathons={hackathons}
-    navbarItems={LIVESITE_NAVBAR}
+import {
+  getActiveHackathon,
+  getHackathons,
+  updateAnnouncement,
+  deleteAnnouncement,
+  subscribeToLivesiteAnnouncements,
+} from '../../utility/firebase';
+import { EDIT, NEW, LIVESITE_NAVBAR } from '../../constants';
+import { useAuth } from '../../utility/auth';
+
+const Markdown = ({ content }) => (
+  <ReactMarkdown
+    linkTarget="_blank"
+    allowedTypes={[
+      'text',
+      'paragraph',
+      'strong',
+      'emphasis',
+      'link',
+      'break',
+      'list',
+      'listItem',
+    ]}
   >
-    <img
-      src="https://i.imgur.com/lEjmB7U.gif"
-      width="480"
-      height="362"
-      alt="panik"
-    />
-    <p>Announcements - This area under construction.</p>
-  </Page>
+    {content}
+  </ReactMarkdown>
 );
 
+const announcementDateFormat = (timestamp) => {
+  const options = {
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+    hour: '2-digit',
+    minute: '2-digit',
+  };
+  return new Date(timestamp).toLocaleDateString('en-US', options);
+};
+
+export default ({ hackathons }) => {
+  const [announcements, setAnnouncements] = useState([]);
+  const [activeHackathon, setActiveHackathon] = useState('');
+  const [isLoading, setIsLoading] = useState(true);
+  const [activeModal, setActiveModal] = useState('');
+  const [currentAnnouncement, setCurrentAnnouncement] = useState({});
+  const { email: user } = useAuth().user;
```

Comment:
what does this do

Reply:
Some black magic giulio wrote in the feature flags component

Reply:
gets the logged in user from firebase auth